Source/Core/Core/AchievementManager.cpp

```diff
@@ -242,6 +252,41 @@ void AchievementManager::DoFrame()
   }
 }
 
+bool AchievementManager::CanPause()
+{
+  u32 frames_to_next_pause = 0;
+  bool can_pause = rc_client_can_pause(m_client, &frames_to_next_pause);
+  if (!can_pause)
+  {
+    OSD::AddMessage("Cannot spam pausing in hardcore mode.", OSD::Duration::VERY_LONG,
+                    OSD::Color::RED);
+    OSD::AddMessage(
+        fmt::format("Can pause in {} seconds.", static_cast<float>(frames_to_next_pause) / 60),
+        OSD::Duration::VERY_LONG, OSD::Color::RED);
+  }
+  return can_pause;
+}
+
+void AchievementManager::DoIdle()
+{
+  std::thread([this]() {
+    while (true)
+    {
+      Common::SleepCurrentThread(1000);
+      std::lock_guard lg{m_lock};
+      if (!m_system || Core::GetState(*m_system) != Core::State::Paused)
+        return;
+      if (!m_background_execution_allowed)
+        return;
+      Core::QueueHostJob([this](Core::System& system) {
+        if (!m_client || !IsGameLoaded())
+          return;
+        rc_client_idle(m_client);
+      });
```

Comment:
While I do agree that QueueHostJob is probably not a bad idea, this is not how you do it. Think about the threading implications here: Your lock from the local thread does not propagate to the host job. The `return;` here will also now no longer stop the local thread, because it just exits the inner lambda.